app/backend/core/authentication.py

```diff
@@ -237,3 +257,74 @@ async def check_path_auth(self, path: str, auth_claims: dict[str, Any], search_c
             break
 
         return allowed
+
+    # See https://github.com/Azure-Samples/ms-identity-python-on-behalf-of/blob/939be02b11f1604814532fdacc2c2eccd198b755/FlaskAPI/helpers/authorization.py#L44
+    async def validate_access_token(self, token: str):
+        """
+        Validate an access token is issued by Entra
+        """
+        jwks = None
+        async for attempt in AsyncRetrying(
+            retry=retry_if_exception_type(AuthError),
+            wait=wait_random_exponential(min=15, max=60),
+            stop=stop_after_attempt(5),
+        ):
+            with attempt:
+                async with aiohttp.ClientSession() as session:
+                    async with session.get(url=self.key_url) as resp:
+                        resp_status = resp.status
+                        if resp_status in [500, 502, 503, 504]:
+                            raise AuthError(
+                                error=f"Failed to get keys info: {await resp.text()}", status_code=resp_status
+                            )
+                        jwks = await resp.json()
+
+        if not jwks or "keys" not in jwks:
```

Comment:
I'm trying to parse the rest of the differences between your code and the sample code, is it mostly additional error message raising? Anything else notable?

Reply:
There's 1 big change: I didn't realize we forgot to set the requested token version in the manifest to 2. Because we didn't do that, we are allowing v1 token issuer/audience in addition to v2. auth_init.py will fix this for folks running it again, but I wanted to make sure it would work if customers changed the code without rerunning provision hooks.

I also changed retries to use tenacity. Other than that they are the same